### PR TITLE
Add: BigLake Metastore API Support to Terraform

### PR DESCRIPTION
**Pull request by nevzheng**
Add: BigLake Metastore API Support to Terraform

If this PR is for Terraform, I acknowledge that I have:

- [] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
biglake: Add Catalog Resource
biglake: Add Database Resource
biglake: Add Table Resource
```

### CODE REVIEW

File: mmv1/products/bigquerybiglake/Catalog.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "bigquery_biglake_catalog"
+    primary_resource_id: "default"
+    vars:
+      catalog_id: "my_catalog"
+      location: "US"
+properties:
+  - !ruby/object:Api::Type::String
+    name: "catalog_id"
```
Comment (melinath) [inline mmv1/products/bigquerybiglake/Catalog.yaml:36]: use `name` as the field name for the main id instead to benefit from default behavior

Comment (melinath) [reply mmv1/products/bigquerybiglake/Catalog.yaml:36]:
```suggestion
    name: "name"
```

Comment (melinath) [reply mmv1/products/bigquerybiglake/Catalog.yaml:36]: should be marked immutable like in the docs - the API will return the longer version of the name but we suppress the diff as long as the last part of the long name is the same.

**Comment by nevzheng** [comment]
github.com/GoogleCloudPlatform/magic-modules/pull/8751
github.com/GoogleCloudPlatform/magic-modules/pull/8753
github.com/GoogleCloudPlatform/magic-modules/pull/8754

**Comment by nevzheng** [comment]
github.com/https://github.com/GoogleCloudPlatform/magic-modules/pull/8751
github.com/https://github.com/GoogleCloudPlatform/magic-modules/pull/8753
github.com/https://github.com/GoogleCloudPlatform/magic-modules/pull/8754

**Pull request closed without merging**